sdk/include/opentelemetry/sdk/trace/processor.h

```diff
@@ -0,0 +1,63 @@
+#pragma once
+
+#include <chrono>
+#include <memory>
+#include "opentelemetry/sdk/trace/recordable.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace sdk
+{
+namespace trace
+{
+/**
+ * Span processor allow hooks for span start and end method invocations.
+ *
+ * Built-in span processors are responsible for batching and conversion of
+ * spans to exportable representation and passing batches to exporters.
+ */
+class SpanProcessor
+{
+public:
+  virtual ~SpanProcessor() = default;
+
+  /**
+   * Create a span recordable. This requests a new span recordable from the
+   * associated exporter.
+   * @return a newly initialized recordable
+   */
+  virtual std::unique_ptr<Recordable> MakeRecordable() noexcept = 0;
```

Comment:
Why is this in the interface? Do I need to implement this function in order to implement an exporter?

Reply:
Yes. For non-optimized cases, the exporter can return a `SpanData` instance, which derives from `Recordable`.

There are good descriptions and discussions about this approach in [this PR](https://github.com/open-telemetry/opentelemetry-cpp/pull/44).